### PR TITLE
refactor: keebio/bamfk1

### PR DESCRIPTION
**Pull request by lesshonor**
## Description

- Migration to standard encoder callback/map functionality instead of `action_exec` on "extra keys".
- Data-driven migration for other attributes.
- Broken Shopify image replaced and bootloader instructions improved.
- ~~:warning: Enabled tri-layers on the `via` keymap, since it uses `TL_LOWR`.~~
- :warning: Mousekeys have been enabled at the keyboard level for Configurator users. This wasn't possible on the `via` keymap for size reasons.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/bamfk1/keymaps/via/rules.mk [hunk 1]
```diff
@@ -1,2 +1,5 @@
+TRI_LAYER_ENABLE = yes
```
Comment (zvecr) [inline keyboards/keebio/bamfk1/keymaps/via/rules.mk:1]: This is enabled when via, so is pointless
```suggestion
```

File: keyboards/keebio/bamfk1/keymaps/via/rules.mk [hunk 2]
```diff
@@ -1,2 +1,5 @@
...
+ENCODER_MAP_ENABLE = yes
+MOUSEKEY_ENABLE = no
```
Comment (zvecr) [inline keyboards/keebio/bamfk1/keymaps/via/rules.mk:3]: Its generally wise to keep this enabled where possible, and trim firmware size elsewhere
```suggestion
```

File: keyboards/keebio/bamfk1/info.json [hunk 3]
```diff
@@ -1,38 +1,64 @@
...
             {"pin_a": "C7", "pin_b": "B5"},
             {"pin_a": "D7", "pin_b": "D4"}
         ]
     },
-    "processor": "atmega32u4",
-    "bootloader": "atmel-dfu",
+    "features": {
+        "audio": true,
+        "bootmagic": true,
+        "console": true,
```
Comment (zvecr) [inline keyboards/keebio/bamfk1/info.json:18]:
```suggestion
        "console": false,
```

**Review by zvecr — CHANGES_REQUESTED** [review]
The issue with this, is that it will break via compatibility,

1. updating via config file will break existing firmware
2. flashing this without the updated config will also fall over 

Previously this has been handled by updating the product ID so that it stops `1` above.

**Comment by nooges** [comment]
@lesshonor Sorry I didn't see this earlier, I have all my QMK GitHub emails filtered to a different folder since I get a ton of them as a contributor. I'm okay if you want to change the PID for the BAMFK-1 and adjust whatever is needed, followed by a "Rev. 1a" VIA definition .json. If you do open a new PR, please ping me on Discord with a link for review, but I have also tried adjusting my email filter settings to try to capture GitHub mentions of my username.

**Comment by nooges** [comment]
I've created a new PR for the migration to encoder map and RGB matrix now: https://github.com/qmk/qmk_firmware/pull/23236

**Pull request closed without merging**